library/src/main/java/io/ona/kujaku/comparisons/Comparison.java

```diff
@@ -0,0 +1,9 @@
+package io.ona.kujaku.comparisons;
+
+public interface Comparison {
+    String TYPE_STRING = "string";
+
+    boolean compare(String a, String type, String b);
```

Comment:
Do we need `String type` argument any more?

Reply:
I think it's good for us to maintain it for easy extensibility in case anyone else needs to add other Comparisons directly from [here](https://github.com/OpenSRP/opensrp-client-native-form/tree/master/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/comparisons)

Reply:
Fair enough, and that's one way of doing it.

However, this approach delegates the responsibility of determining `type` to the calling function, right? 

Another approach may be letting the compare method impute the type from the `String a` and `String b` arguments thus having a central location for determining type and transferring that responsibility from the calling function(s)?

In this case, would we need the `type` argument?

Reply:
> However, this approach delegates the responsibility of determining `type` to the calling function, right?

Yeap, this is OK by me, I want the developer to explicitly do this and the calling function to have the responsibility to do that. Just as in the native forms

Reply:
OK.